webapp/src/modules/wallet/utils.ts

```diff
@@ -47,10 +48,12 @@ export function formatBalance(balance: number) {
 export function* waitForWalletConnectionAndIdentityIfConnecting() {
   const isConnectingToWallet = (yield select(isConnecting)) as ReturnType<typeof isConnecting>
   if (isConnectingToWallet) {
-    const { success } = (yield race({
+    const { success, timeout } = (yield race({
       success: take(CONNECT_WALLET_SUCCESS),
-      failure: take(CONNECT_WALLET_FAILURE)
-    })) as { success: ConnectWalletSuccessAction; failure: ConnectWalletFailureAction }
+      failure: take(CONNECT_WALLET_FAILURE),
+      timeout: delay(WAIT_FOR_WALLET_CONNECTION_TIMEOUT) // 10 seconds timeout
+    })) as { success: ConnectWalletSuccessAction; failure: ConnectWalletFailureAction; timeout: any }
```

Comment:
Would you mind typing the return of delay? We don't have the linter rule available for it yet, but typing it would help on the changes we need to do:
```suggestion
    })) as { success: ConnectWalletSuccessAction; failure: ConnectWalletFailureAction; timeout: undefined }
```
If I'm not wrong, it should return undefined